gax-grpc/src/test/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProviderTest.java

```diff
@@ -385,4 +393,78 @@ public void testWithPrimeChannel() throws IOException {
           .primeChannel(Mockito.any(ManagedChannel.class));
     }
   }
+
+  @Test
+  public void testWithDefaultDirectPathServiceConfig() {
+    InstantiatingGrpcChannelProvider provider =
+        InstantiatingGrpcChannelProvider.newBuilder().build();
+
+    ImmutableMap<String, ?> defaultServiceConfig = provider.directPathServiceConfig;
+
+    List<Map<String, ?>> lbConfigs = getAsObjectList(defaultServiceConfig, "loadBalancingConfig");
+    assertThat(lbConfigs).hasSize(1);
+    Map<String, ?> lbConfig = lbConfigs.get(0);
+    Map<String, ?> grpclb = getAsObject(lbConfig, "grpclb");
+    List<Map<String, ?>> childPolicies = getAsObjectList(grpclb, "childPolicy");
+    assertThat(childPolicies).hasSize(1);
+    Map<String, ?> childPolicy = childPolicies.get(0);
+    assertThat(childPolicy.keySet()).containsExactly("pick_first");
+  }
+
+  @Nullable
+  private static Map<String, ?> getAsObject(Map<String, ?> json, String key) {
+    Object mapObject = json.get(key);
+    if (mapObject == null) {
+      return null;
+    }
+    return checkObject(mapObject);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static Map<String, ?> checkObject(Object json) {
+    checkArgument(json instanceof Map, "Invalid json object representation: %s", json);
+    for (Map.Entry<Object, Object> entry : ((Map<Object, Object>) json).entrySet()) {
+      checkArgument(entry.getKey() instanceof String, "Key is not string");
+    }
+    return (Map<String, ?>) json;
+  }
+
+  private static List<Map<String, ?>> getAsObjectList(Map<String, ?> json, String key) {
+    Object listObject = json.get(key);
+    if (listObject == null) {
+      return null;
+    }
+    return checkAndJsonListOfObjects(listObject);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static List<Map<String, ?>> checkAndJsonListOfObjects(Object listObject) {
```

Comment:
this doesn't sound right: check*And*JsonListOfObjects?

Reply:
oops i was debating checkAndGet or check, somehow the and survived. thanks!